vendor/clink.lua

```diff
@@ -403,17 +410,21 @@ local function git_prompt_filter()
           local branch = get_git_branch(git_dir)
           local color
           if branch then
-              -- Has branch => therefore it is a git folder, now figure out status
-              local gitStatus = get_git_status()
-              local gitConflict = get_git_conflict()
+              if cmderGitStatusOptIn ~= 'branchonly' then
```

Comment:
The [vladimir-kotikov/clink-completions](https://github.com/vladimir-kotikov/clink-completions) repo has a dependency on `cmderGitStatusOptIn` being a true/false value.  The contents of that repo are copied into the %CMDER_ROOT%\vendor\clink-completions directory.

This new setting won't behave as intended in certain repos, and when it's used it will bring back the slow behavior https://github.com/cmderdev/cmder/issues/2484 that was fixed by https://github.com/cmderdev/cmder/commit/6027ac31c428a3b9a5cf12424d4d95787e279023.

Reply:
Thanks, wasn't aware of this.

Would the following seem OK instead?

```
local statusSetting = get_git_status_setting()
cmderGitStatusOptIn = statusSetting == true
...
if statusSetting ~= 'branchonly' then
```

I'll be unavailable for the next few days but can come back and iterate on this next week.

Reply:
Although.. clink-completions just does `cmderGitStatusOptIn == false` so `'branchonly' == false` would be false, which may actually be what we want here anyway. I could still do `cmderGitStatusOptIn = statusSetting ~= false` to make this more explicit on our end.

But my question on that is... do we want clink-completions's filter to still run. The goal of this PR was to avoid expensive operations, and the clink-completions filter appears to only be doing fast operations that aren't affected by repo size.

Reply:
I am not aware of anyone ever complaining we showed the Git branch in the prompt.  The only complaint has ALWAYS been the prompt can be slow in a large Git repo.  

My thought was that `*status = false` changes from 'no status and no branch' to 'no status with branch' so we scrap the `cmdstatus = branchonly` setting.

Currently Cmder's Git Opt Out functionality has feature parity across `Cmd`, `Powershell`, and `Bash` shells this PR only affects git status opt out in `Cmd` shells so we should also:

- Modify `%cmder_root%`\vendor\git-prompt.sh  to just display the branch for Bash shells if `status=false` or `bashstatus=false`
- Modify `%cmder_root%`\vendor\psmodules\Cmder.ps1`  to just display the branch for Powershell shells if `status=false` or `psstatus=false`
  -  We do not have much control of how Powershell prompt is displayed because we use a function from `Posh-Git` so this  will be more difficult.

This would maintain current compatibility with `bash` and `powershell` shells for these settings. 

Lastly disabling git status checks in Cmder whether the branch is displayed or not the clink-completions git-prompt filter should not run either.

Reply:
> the clink-completions filter appears to only be doing fast operations that aren't affected by repo size.

It only seems that way because most of us don't use repos large enough to reproduce the problem. I recommend reading #2484, and especially [this comment](https://github.com/cmderdev/cmder/issues/2484#issuecomment-787043078).

Reply:
This all sounds good to me, so I've:
1. Removed the "branchonly" concept
2. Made `cmder.*status=false` only prevent the status operations, not printing branch name for cmd
3. Added new simple branch name logic for bash and powershell to mimic the behavior of cmd in the `status=false` case

Reply:
(And I'm no longer touching anything related to clink-completions so that is still off when `status=false`)

Reply:
The .lua part looks good to me now.  I have no familiarity with the affected .sh or .ps1 scripts or how Cmder integrates with those shells, though.

Reply:
Will test it out later.  Thanks